ext/drreg/drreg.c

```diff
@@ -437,6 +669,73 @@ drreg_event_bb_insert_late(void *drcontext, void *tag, instrlist_t *bb, instr_t
     }
 
     /* Before each app read, or at end of bb, restore spilled registers to app values: */
+    for (reg = DR_REG_APPLICABLE_START_SIMD; reg <= DR_REG_APPLICABLE_STOP_SIMD; reg++) {
```

Comment:
This is duplicating the loop for GPRs right? And then doing the "same", but everything "indirectly" for simds. The problem I see here is that it becomes a lot of duplicated code, and it becomes very hard to read. Do you think it is possible to re-factor this in a way that you have the common code in functions, and the differentiation of the restore and spill in functions as well?

Reply:
I was going to do this, making use of a lot of function pointer usage to differentiate between gpr and sims, but this would result in heavy refactoring of code. I was worried that the PR would be too different from the original and make things more complex. Perhaps we can add an issue specifically for this case, as like I said, some refactoring would be in order.

Reply:
I think the incremental refactoring to prepare would be better than adding lots of redundant code and clean up afterwards. WDYT?

Reply:
If you agree, I think this will be time very well spent!

Reply:
Yes, I agree :thumbsup:

Reply:
Hi John, this patch here is becoming more important to us now because it is gating a few things we are also committed to work on. Your contribution is very appreciated, are you working on this and what are your plans?

Reply:
Hey @hgreving2304 yes I plan to work on it on an incremental bases as fast as I can. Unfortunately, I cannot dedicate a lot of time at a stretch as I am busy on other matters at the moment ( I hope you understand). When you say this patch, are you referring to this particular comment, or in general?

Reply:
> to this particular comment, or in general?

This patch in general. I used this comment to reply since it mentions a new dependency to re-factor the code to avoid the duplication. Are you able to share what your plans are in terms of time? Would it help if we help you to re-factor and update upstream so you can merge?

Reply:
I am currently swamped till Thursday, but I'll see what I can do later on tonight.

Reply:
> Would it help if we help you to re-factor and update upstream so you can merge?

Feel free! Thanks

Reply:
I'll focus on this in a few hours. Sorry for the delay.